portfolio/src/main/webapp/script.js

```diff
@@ -56,6 +56,51 @@ function getComments() {
     });
 }
 
+// function to create comment form
+function createCommentForm() {
+    fetch('/login').then(response => response.json()).then(loginDetails => {
+        const logButtonContainer = document.getElementById('log-button');
+        const logLabel = document.createElement('label');
+        
+        // display button to log in
+        if (!loginDetails.loggedIn) {
+            // create login button
+            logLabel.innerText = "Please login to post comments.";
+            logButton = buttonHref("Log In", loginDetails.loginUrl);
+        }
+
+        // display comment form
+        else {
+            // create logout button
+            logLabel.innerText = "You are now logged in as " + loginDetails.userEmail + ".";
+            logButton = buttonHref("Log Out", loginDetails.logoutUrl);
+
+            // create form
+            const commentFormContainer = document.getElementById('comment-form');
```

Comment:
Using javascript to create comment-form is fine.

Alternatively, you can hide the comment-form in HTML by default, and show it later.
This way you don't have to manually construct complex Html tags in Javascript.

Reply:
ok, noted. thanks! it makes it much simpler